README.md

```diff
@@ -1,89 +1,218 @@
+
 ## Orchestrator
 
-WIP
+The Orchestrator in the Summa Aggregation serves as the central management component, coordinating the data processing activities. It plays a pivotal role in coordinating the activities of `Executors` and `Workers`, improving of tasks in the generation of Merkle sum tree.
+Key functions of the `Orchestrator` include:
+
+- **Dynamic Executor Spawning**: The Orchestrator dynamically spawns `Executors` in numbers set by the user. Each `Executor` is then connected to a dedicated `Worker` for efficient task execution.
+
+- **Task Management and Distribution**: It oversees the overall task flow, loading tasks and distributing them to `Executors`.
+
+- **Error Management and Pipeline Control**: The Orchestrator handles basic pipeline control and responds to errors by initiating the cancellation of all tasks.
+
+## Executor and Worker
+
+The `Executor` acts as a crucial intermediary between the `Orchestrator` and `Workers`, facilitating the data processing workflow. Spawned by the `Orchestrator`, each `Executor` operates in a one-to-one relationship with a `Worker`. The `Worker` is a server that can generate a merkle sum tree, internally called a `mini-tree`, by receiving entries data. The primary role of a `Worker` is to build these `mini-trees`, which are segments of the `AggregationMerkleSumTree`.
+
+Key aspects of the `Executor's` role include:
+
+- **Spawning and Connection**: `Executors` are dynamically spawned by the `Orchestrator` as part of the system's scalability. Each `Executor` is designed to connect with a `Worker` for task execution.
+
+- **Data Handling and Task Distribution**: A primary function of the `Executor` is to receive data entries, often parsed and prepared by the Orchestrator. Upon receiving these entries, the Executor is responsible for forwarding them to its connected `Worker`.
+
+- **Communication Bridge**: The `Executor` serves as a communication bridge within the data pipeline. It relays processed data, `mini-trees`, from Workers back to the Orchestrator.
+
+The `Worker` called in here, mostly point out the container that runs `mini-tree-generator` server.
+
+## ExecutorSpawner
+
+The `ExecutorSpawner` is responsible for initializing and terminating `Executors`. It manages the creation of `Executor` instances and `Workers`, with the latter serving as the `mini-tree-generator` accessible by the `Executor`.
+
+In the Summa-Aggregation, there are three types of `ExecutorSpawner`:
+
+- **MockSpawner**: Primarily used for testing, this spawner initializes `Executors` suitable for various test scenarios, including negative test cases. The Worker spawned by this spawner runs a server locally.
+
+- **LocalSpawner**: It is close to actual use cases, this spawner enables users to initialize `Executors` and `Workers` in local Docker environments.
+
```

Comment:
I can say that the `LocalSpawner` creates workers with each container acting as a unit, and these containers run on a host (the main machine).
 On the other hand, `CloudSpawner` creates workers by using services as units. It can adjust the number of workers through the `scale` variable, and these services (consisting of containers) run on different hosts (nodes, outside of the main machine).

These approaches may seem similar, but from the perspective of Docker management, they are quite different.